### PR TITLE
schedules

### PR DESCRIPTION
**Pull request by jmonster**
+ [x] layout how schedules work in the `README`
+ [x] define structures for describing and storing schedules
+ [x] implement a function that returns the currently schedule temperature
+ [x] load schedules from EEPROM in `setup()`
+ [x] export a cloud function to allow a server to send an updated schedule
+ [x] document how to submit an updated schedule
+ [x] support overriding a scheduled temperture

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,20 +11,73 @@ If your application contains multiple files, they should all be included in the
...
+#define WEEKDAY =
+	MONDAY | TUESDAY | WEDNESDAY | THURSDAY | FRIDAY;
+
+#define WEEKEND =
+	SATURDAY | SUNDAY;
+
+struct ScheduleEvent {
+	char daysMask;          // Mo Tu We Th Fr Sa Sn
+  char timeInterval;      // multiple of 15; 00:00 to 23:45
+  char temperatureOffset; // 0 - 255; 50 F to 113.75 F
```
Comment (tupakapoor) [inline README.md]: BASE_TEMP is in celsius - this is in fahrenheit (unsure if it actually matters?) - just want to make sure we're being consistent throughout as that will reduce confusion going forward. I assume UNITS_PER_DEGREE is also in celsius? But see I'm already confused 😛

File: README.md [hunk 2]
```diff
@@ -17,20 +11,73 @@ If your application contains multiple files, they should all be included in the
...
+#define COMFORTABLE_TEMP_CELSIUS =
+	20.5; // ~69 Fahrenheit
+#define COMFORTABLE_TEMP_AS_OFFSET =
+	(COMFORTABLE_TEMP_CELSIUS - BASE_TEMP) * UNITS_PER_DEGREE;
+
+// 69º F on weekdays at 7AM
+struct ScheduleEvent weekdays = { WEEKDAY, SEVEN_AM, COMFORTABLE_OFFSET };
+
+// 7 day schedule
+struct ScheduleEvent events[] = [weekdays;
```
Comment (tupakapoor) [inline README.md]: missing a `]` after weekdays

File: README.md [hunk 3]
```diff
@@ -17,20 +11,73 @@ If your application contains multiple files, they should all be included in the
...
+```
+
+#### Persisting a schedule
+* store ScheduleEvent objects sequentially.
```
Comment (tupakapoor) [inline README.md]: Sequentially meaning in chronological order?

File: src/Schedule.cpp [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  // scan for events active today
+
+  // scan for most recent event that's still in the past
+
+  // return the params for that event
+
+  // Serial.println(dayAsMask);
+
+  unsigned char isTuesday = (bool)(dayAsMask & TUESDAY);
+  Serial.println(isTuesday);
```
Comment (tupakapoor) [inline src/Schedule.cpp]: debugging?

File: README.md [hunk 5]
```diff
@@ -17,20 +11,70 @@ If your application contains multiple files, they should all be included in the
...
+#define THURSDAY  1 << 4 // 2^4
+#define FRIDAY    1 << 5 // 2^5
+#define SATURDAY  1 << 6 // 2^6
+
+#define WEEKDAY (MONDAY | TUESDAY | WEDNESDAY | THURSDAY | FRIDAY)
+#define WEEKEND (SATURDAY | SUNDAY)
+
+struct ScheduleEvent {
+	char daysMask;                   // Mo Tu We Th Fr Sa Sn
+  char timeInterval;               // multiple of 15; 00:00 to 23:45
```
Comment (tupakapoor) [inline README.md]: what up with this indentation
get your shit together

File: src/Thermal.cpp [hunk 6]
```diff
@@ -65,14 +67,22 @@ void setup() {
...
+
+  Serial.begin(9600);
+
   while (!bmp.begin(1)) {
     delay(2000);
   }
 }
 
 void loop() {
+  Serial.println(Time.now());
```
Comment (tupakapoor) [inline src/Thermal.cpp]: Leaving console.log's in your PR's now eh buddy?

Comment (jmonster) [inline src/Thermal.cpp]: this means I have free license to 💩 all over the PR 🙃 I'll be overwriting the commit.

File: README.md (hunk 3 above)
Comment (jmonster) [inline README.md]: I went ahead and just rewrote it to not require a sorted list 🙃

**Pull request merged**